File: routes/GET/getEpisodes.py
from classes import episodes
from flask import make_response, jsonify

def get(animeId):
    try:
        episode = episodes.Episode(animeId=animeId)
        getEpisodes =episode.getAllEpisodes() 

        match getEpisodes["message"]:
            case "The given anime doesn't exist":
                resp= make_response(jsonify(getEpisodes))
                resp.status_code = 404
                return resp
            case "episodes found":
                resp= make_response(jsonify(getEpisodes))
                resp.status_code = 200
                return resp
            case "An error has occurred while getting episodes":
                resp= make_response(jsonify(getEpisodes))
                resp.status_code = 500
                return resp

    except Exception as e:
        resp = make_response(jsonify({"message":"There was an error", "error":e.args}))
        resp.status_code= 500
        return resp

File: src/routes/GET/getEpisodes.ts
import type { Response, Request } from "express";
import Episode from "../../classes/episodes";

export default async function getEpisodes(req: Request, res: Response){
  try {
    const animeId = Number(req.params.animeId);
    const episodes = await new Episode(0,"",0,"","","",animeId).getAll();

    switch (episodes.message) {
      case "The anime does not exist":
        res.statusCode = 404;
        res.json(episodes);
        break;
      case "success":
        res.statusCode = 200;
        res.json(episodes);
        break;
      case "An error has occurred while getting the episodes":
        res.statusCode = 500;
        res.json(episodes);
        break;
    }
  } catch (error: any) {
    res.statusCode = 500;
    res.json({ message: "There was an error", error: error.message });
  }
}
